packages/client-sdk-nodejs/src/internal/storage-control-client.ts

```diff
@@ -0,0 +1,159 @@
+import {control} from '@gomomento/generated-types';
+import grpcControl = control.control_client;
+import {Header, HeaderInterceptorProvider} from './grpc/headers-interceptor';
+import {ClientTimeoutInterceptor} from './grpc/client-timeout-interceptor';
+import {Status} from '@grpc/grpc-js/build/src/constants';
+import {CacheServiceErrorMapper} from '../errors/cache-service-error-mapper';
+import {ChannelCredentials, Interceptor} from '@grpc/grpc-js';
+import {MomentoLogger, StoreInfo, ListStores} from '..';
+import {version} from '../../package.json';
+import {IdleGrpcClientWrapper} from './grpc/idle-grpc-client-wrapper';
+import {GrpcClientWrapper} from './grpc/grpc-client-wrapper';
+import {validateStoreName} from '@gomomento/sdk-core/dist/src/internal/utils';
+import {CreateStore, DeleteStore} from '@gomomento/sdk-core';
+import {StorageClientPropsWithConfig} from './storage-client-props-with-config';
+
+export class StorageControlClient {
+  private readonly clientWrapper: GrpcClientWrapper<grpcControl.ScsControlClient>;
+  private readonly interceptors: Interceptor[];
+  private static readonly REQUEST_TIMEOUT_MS: number = 60 * 1000;
+  private readonly logger: MomentoLogger;
+  private readonly cacheServiceErrorMapper: CacheServiceErrorMapper;
+
+  /**
+   * @param {StorageClientProps} props
+   */
+  constructor(props: StorageClientPropsWithConfig) {
+    this.logger = props.configuration.getLoggerFactory().getLogger(this);
+    this.cacheServiceErrorMapper = new CacheServiceErrorMapper(false);
+    const headers = [
+      new Header('Authorization', props.credentialProvider.getAuthToken()),
+      new Header('Agent', `nodejs:store:${version}`),
```

Comment:
is this header getting sent up on every request

Reply:
No, the Header Interceptor class takes care of that:
https://github.com/momentohq/client-sdk-javascript/blob/main/packages/client-sdk-nodejs/src/internal/grpc/headers-interceptor.ts

Reply:
got it thank you!